code/modules/projectiles/gun_attachables/scope.dm

```diff
@@ -0,0 +1,243 @@
+	icon_state = "sniperscope_invisible"
+	zoom_viewsize = 0
+	zoom_tile_offset = 5
+	deployed_scope_rezoom = TRUE
+
+//all mounted guns with a nest use this
+/obj/item/attachable/scope/unremovable/hsg_102/nest
+	scope_delay = 2 SECONDS
+	zoom_tile_offset = 7
+	zoom_viewsize = 2
+	deployed_scope_rezoom = FALSE
+
+/obj/item/attachable/scope/activate(mob/living/carbon/user, turn_off)
+	if(turn_off)
+		if(SEND_SIGNAL(user, COMSIG_ITEM_ZOOM) &  COMSIG_ITEM_ALREADY_ZOOMED)
+			zoom(user)
+		return TRUE
+
+	if(!(master_gun.item_flags & WIELDED) && !CHECK_BITFIELD(master_gun.item_flags, IS_DEPLOYED))
+		if(user)
+			to_chat(user, span_warning("You must hold [master_gun] with two hands to use [src]."))
+		return FALSE
+	if(CHECK_BITFIELD(master_gun.item_flags, IS_DEPLOYED) && user.dir != master_gun.loc.dir)
+		user.setDir(master_gun.loc.dir)
+	if(!do_after(user, scope_delay, NONE, src, BUSY_ICON_BAR))
+		return FALSE
+	zoom(user)
+	update_icon()
+	return TRUE
+
+/obj/item/attachable/scope/zoom_item_turnoff(datum/source, mob/living/carbon/user)
+	if(ismob(source))
+		INVOKE_ASYNC(src, PROC_REF(activate), source, TRUE)
+	else
+		INVOKE_ASYNC(src, PROC_REF(activate), user, TRUE)
+
+/obj/item/attachable/scope/onzoom(mob/living/user)
+	if(zoom_allow_movement)
+		user.add_movespeed_modifier(MOVESPEED_ID_SCOPE_SLOWDOWN, TRUE, 0, NONE, TRUE, zoom_slowdown)
+		RegisterSignal(user, COMSIG_LIVING_SWAPPED_HANDS, PROC_REF(zoom_item_turnoff))
+	else
+		RegisterSignals(user, list(COMSIG_MOVABLE_MOVED, COMSIG_LIVING_SWAPPED_HANDS), PROC_REF(zoom_item_turnoff))
+	if(!CHECK_BITFIELD(master_gun.item_flags, IS_DEPLOYED))
+		RegisterSignal(user, COMSIG_MOB_FACE_DIR, PROC_REF(change_zoom_offset))
+	RegisterSignals(master_gun, list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_UNWIELD, COMSIG_ITEM_DROPPED), PROC_REF(zoom_item_turnoff))
+	master_gun.accuracy_mult += scoped_accuracy_mod
+	if(has_nightvision)
+		update_remote_sight(user)
+		user.reset_perspective(src)
+		active_nightvision = TRUE
+
+/obj/item/attachable/scope/onunzoom(mob/living/user)
+	if(zoom_allow_movement)
+		user.remove_movespeed_modifier(MOVESPEED_ID_SCOPE_SLOWDOWN)
+		UnregisterSignal(user, list(COMSIG_LIVING_SWAPPED_HANDS, COMSIG_MOB_FACE_DIR))
+	else
+		UnregisterSignal(user, list(COMSIG_MOVABLE_MOVED, COMSIG_LIVING_SWAPPED_HANDS, COMSIG_MOB_FACE_DIR))
+	UnregisterSignal(master_gun, list(COMSIG_ITEM_EQUIPPED, COMSIG_ITEM_UNWIELD, COMSIG_ITEM_DROPPED))
+	master_gun.accuracy_mult -= scoped_accuracy_mod
+	if(has_nightvision)
+		user.update_sight()
+		user.reset_perspective(user)
+		active_nightvision = FALSE
+
+/obj/item/attachable/scope/update_remote_sight(mob/living/user)
+	. = ..()
+	user.see_in_dark = 32
+	user.lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+	user.sync_lighting_plane_alpha()
+	return TRUE
+
+/obj/item/attachable/scope/zoom(mob/living/user, tileoffset, viewsize)
```

Comment:
I assume it was already like this, but new types should be defined after procs of the parent type.

Reply:
Fixed